operator/pkg/gateway-api/cell.go

```diff
@@ -51,13 +53,17 @@ var Cell = cell.Module(
 	cell.Provide(registerSecretSync),
 )
 
-var requiredGVK = []schema.GroupVersionKind{
+var requiredGVKs = []schema.GroupVersionKind{
 	gatewayv1.SchemeGroupVersion.WithKind("gatewayclasses"),
 	gatewayv1.SchemeGroupVersion.WithKind("gateways"),
 	gatewayv1.SchemeGroupVersion.WithKind("httproutes"),
 	gatewayv1.SchemeGroupVersion.WithKind("grpcroutes"),
 	gatewayv1beta1.SchemeGroupVersion.WithKind("referencegrants"),
+}
+
+var optionalGVKs = []schema.GroupVersionKind{
 	gatewayv1alpha2.SchemeGroupVersion.WithKind("tlsroutes"),
```

Comment:
would it be worth to extract consts for the kind names that then could also be referenced from the other places that are conditionally enable functionality if the CRD is present? (otherwise these strings are spread all over the code).

Reply:
Yes, agreed.